k-distribution/tests/regression-new/set-symbolic-tests/non-terminating/diff-21-spec.k

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) 2022 K Team. All Rights Reserved.
+// generated by gen-tests.sh
+
+requires "set-tests.k"
+
+module DIFF-21-SPEC
+    imports SET-TESTS
+
+    claim <k> diff ( Z:MyId in ( (S SetItem(X:MyId)) -Set SetItem(Y:MyId) ) )     => . </k> requires Y  ==K X
```

Comment:
All non-terminating tests are also looping when running against a version without any new set simplification rules.
The problematic terms that make the tests loop are `S SetItem(X) -Set SetItem(Y)` and `SetItem(Y) |Set (S SetItem(X))`. I added this information to the [relevant ticket  in `haskell-backend`](https://github.com/runtimeverification/haskell-backend/issues/2928)

Reply:
Confirmed that these tests in `non-terminating` work when using a HS backend without internal set simplification (from PR https://github.com/runtimeverification/haskell-backend/pull/3062)